pkg/agent/ipassigner/linkmonitor/link_monitor_linux.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkmonitor
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/set"
+
+	utilnetlink "antrea.io/antrea/pkg/agent/util/netlink"
+)
+
+var linkSubscribeFunc = netlink.LinkSubscribeWithOptions
+
+type linkMonitor struct {
+	mutex         sync.RWMutex
+	cacheSynced   bool
+	eventHandlers map[string][]LinkEventHandler
+	linkNames     set.Set[string]  // known link names
+	linkIndexMap  map[int32]string // map from link index to link name
+	netlink       utilnetlink.Interface
+}
+
+func NewLinkMonitor() *linkMonitor {
+	return &linkMonitor{
+		eventHandlers: make(map[string][]LinkEventHandler),
+		linkNames:     set.New[string](),
+		linkIndexMap:  make(map[int32]string),
+		netlink:       &netlink.Handle{},
+	}
+}
+
+func (d *linkMonitor) HasSynced() bool {
+	d.mutex.RLock()
+	defer d.mutex.RUnlock()
+	return d.cacheSynced
+}
+
+func (d *linkMonitor) AddEventHandler(handler LinkEventHandler, linkNames ...string) {
+	if len(linkNames) == 0 {
+		d.eventHandlers[""] = append(d.eventHandlers[""], handler)
```

Comment:
define a global constant `linkAny = ""` to improve readability

Reply:
added constant linkAny.